keras_nlp/models/falcon/falcon_backbone.py

```diff
@@ -0,0 +1,164 @@
+        self,
+        vocabulary_size,
+        num_layers,
+        num_attention_heads,
+        hidden_dim,
+        intermediate_dim,
+        layer_norm_epsilon=1e-5,
+        attention_dropout=0,
+        feedforward_dropout=0,
+        dtype=None,
+        **kwargs,
+    ):
+        # === Layers ===
+        # Embed Tokens
+        token_embedding_layer = ReversibleEmbedding(
+            input_dim=vocabulary_size,
+            output_dim=hidden_dim,
+            dtype=dtype,
+            name="token_embedding",
+        )
+
+        # Apply successive transformer decoder blocks.
+        transformer_layers = []
+        for i in range(num_layers):
+            layer = FalconTransformerDecoder(
+                num_attention_heads=num_attention_heads,
+                intermediate_dim=intermediate_dim,
+                attention_dropout=attention_dropout,
+                feedforward_dropout=feedforward_dropout,
+                dtype=dtype,
+                name=f"transformer_layer_{i}",
+            )
+            transformer_layers.append(layer)
+
+        final_layernorm = keras.layers.LayerNormalization(
+            epsilon=layer_norm_epsilon,
+            dtype=dtype,
+            name="final_layernorm",
+        )
+
+        # === Functional Model ===
+        token_ids = keras.Input(shape=(None,), dtype="int32", name="token_ids")
+        padding_mask = keras.Input(
+            shape=(None,), dtype="int32", name="padding_mask"
+        )
+        x = token_embedding_layer(token_ids)
+
+        for transformer_layer in transformer_layers:
+            x = transformer_layer(inputs=x, decoder_padding_mask=padding_mask)
+        sequence_output = final_layernorm(x)
+
+        super().__init__(
+            inputs={
+                "token_ids": token_ids,
+                "padding_mask": padding_mask,
+            },
+            outputs=sequence_output,
+            **kwargs,
+        )
+
+        # === Config ===
+        self.vocabulary_size = vocabulary_size
+        self.num_layers = num_layers
+        self.num_attention_heads = num_attention_heads
+        self.hidden_dim = hidden_dim
+        self.intermediate_dim = intermediate_dim
+        self.attention_dropout = attention_dropout
+        self.feedforward_dropout = feedforward_dropout
+        self.layer_norm_epsilon = layer_norm_epsilon
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "vocabulary_size": self.vocabulary_size,
+                "num_layers": self.num_layers,
+                "num_attention_heads": self.num_attention_heads,
+                "hidden_dim": self.hidden_dim,
+                "intermediate_dim": self.intermediate_dim,
+                "attention_dropout": self.attention_dropout,
+                "feedforward_dropout": self.feedforward_dropout,
+                "layer_norm_epsilon": self.layer_norm_epsilon,
+            }
+        )
+        return config
+
+    @property
```

Comment:
With above assigning to self, this can be removed.

Reply:
Done.